smpp/server.go

```diff
@@ -178,19 +185,20 @@ func (srv *server) Serve() {
 // handle new clients.
 func (srv *server) handle(c *conn) {
 	defer c.Close()
-	if err := srv.auth(c); err != nil {
-		if err != io.EOF {
-			log.Println("Server auth failed:", err)
-		}
-		return
-	}
+
 	// Use connSwitch to have synced read/write
 	s := &session{conn: &connSwitch{}}
 	s.conn.Set(c)
 	s.id = randomString(IDLen)
 	srv.mu.Lock()
 	srv.s[s.id] = s
 	srv.mu.Unlock()
+
+	if err := srv.auth(c, s); err != nil {
+		log.Println("Server auth failed:", err)
+		return
```

Comment:
What's the expected behaviour when the auth fails, maybe closing the connection, sending some error to the client ?